### PR TITLE
[AMORO-1985]Avoid NPE during disposing ArcticServiceContainer

### LINKED ISSUE #1985: [Bug]: Avoid NPE during disposing ArcticServiceContainer
### What happened?

Before calling the cancel method of `com.netease.arctic.server.DefaultOptimizingService#optimizerMonitorTimer`, we should first check whether it is null to avoid the following NullPointerException. We don't need to worry about the InvalidClassException because it occurred due to the lack of updating the database after the upgrade.

### Affects Versions

0.5.0

### What engines are you seeing the problem on?

AMS

### How to reproduce

_No response_

### Relevant log output

```shell
2023-09-14 15:29:19,314 ERROR [main] [com.netease.arctic.server.ArcticServiceContainer] [] - AMS start error
java.lang.IllegalArgumentException: deserialization error
        at com.netease.arctic.utils.SerializationUtil.simpleDeserialize(SerializationUtil.java:75) ~[amoro-core-0.6.0-SNAPSHOT.jar:?]
        at com.netease.arctic.server.persistence.TaskFilesPersistence.loadTaskInputs(TaskFilesPersistence.java:31) ~[amoro-ams-server-0.6.0-SNAPSHOT.jar:?]
        at com.netease.arctic.server.optimizing.OptimizingQueue$TableOptimizingProcess.loadTaskRuntimes(OptimizingQueue.java:611) ~[amoro-ams-server-0.6.0-SNAPSHOT.jar:?]
        at com.netease.arctic.server.optimizing.OptimizingQueue$TableOptimizingProcess.<init>(OptimizingQueue.java:388) ~[amoro-ams-server-0.6.0-SNAPSHOT.jar:?]
        at com.netease.arctic.server.optimizing.OptimizingQueue.initTableRuntime(OptimizingQueue.java:93) ~[amoro-ams-server-0.6.0-SNAPSHOT.jar:?]
        at java.util.ArrayList.forEach(ArrayList.java:1259) ~[?:1.8.0_291]
        at com.netease.arctic.server.optimizing.OptimizingQueue.<init>(OptimizingQueue.java:86) ~[amoro-ams-server-0.6.0-SNAPSHOT.jar:?]
        at com.netease.arctic.server.DefaultOptimizingService.lambda$loadOptimizingQueues$1(DefaultOptimizingService.java:110) ~[amoro-ams-server-0.6.0-SNAPSHOT.jar:?]
        at java.util.ArrayList.forEach(ArrayList.java:1259) ~[?:1.8.0_291]
        at com.netease.arctic.server.DefaultOptimizingService.loadOptimizingQueues(DefaultOptimizingService.java:104) ~[amoro-ams-server-0.6.0-SNAPSHOT.jar:?]
        at com.netease.arctic.server.DefaultOptimizingService$TableRuntimeHandlerImpl.initHandler(DefaultOptimizingService.java:346) ~[amoro-ams-server-0.6.0-SNAPSHOT.jar:?]
        at com.netease.arctic.server.table.RuntimeHandlerChain.initialize(RuntimeHandlerChain.java:32) ~[amoro-ams-server-0.6.0-SNAPSHOT.jar:?]
        at com.netease.arctic.server.table.DefaultTableService.initialize(DefaultTableService.java:311) ~[amoro-ams-server-0.6.0-SNAPSHOT.jar:?]
        at com.netease.arctic.server.ArcticServiceContainer.startService(ArcticServiceContainer.java:142) ~[amoro-ams-server-0.6.0-SNAPSHOT.jar:?]
        at com.netease.arctic.server.ArcticServiceContainer.main(ArcticServiceContainer.java:106) ~[amoro-ams-server-0.6.0-SNAPSHOT.jar:?]
Caused by: java.io.InvalidClassException: com.netease.arctic.optimizing.RewriteFilesInput; local class incompatible: stream classdesc serialVersionUID = 1106605615053484570, local class serialVersionUID = -6895748366536402812
        at java.io.ObjectStreamClass.initNonProxy(ObjectStreamClass.java:699) ~[?:1.8.0_291]
        at java.io.ObjectInputStream.readNonProxyDesc(ObjectInputStream.java:2001) ~[?:1.8.0_291]
        at java.io.ObjectInputStream.readClassDesc(ObjectInputStream.java:1848) ~[?:1.8.0_291]
        at java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2158) ~[?:1.8.0_291]
        at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1665) ~[?:1.8.0_291]
        at java.io.ObjectInputStream.readObject(ObjectInputStream.java:501) ~[?:1.8.0_291]
        at java.io.ObjectInputStream.readObject(ObjectInputStream.java:459) ~[?:1.8.0_291]
        at java.util.HashMap.readObject(HashMap.java:1412) ~[?:1.8.0_291]
        at sun.reflect.GeneratedMethodAccessor44.invoke(Unknown Source) ~[?:?]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_291]
        at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_291]
        at java.io.ObjectStreamClass.invokeReadObject(ObjectStreamClass.java:1185) ~[?:1.8.0_291]
        at java.io.ObjectInputStream.readSerialData(ObjectInputStream.java:2294) ~[?:1.8.0_291]
        at java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2185) ~[?:1.8.0_291]
        at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1665) ~[?:1.8.0_291]
        at java.io.ObjectInputStream.readObject(ObjectInputStream.java:501) ~[?:1.8.0_291]
        at java.io.ObjectInputStream.readObject(ObjectInputStream.java:459) ~[?:1.8.0_291]
        at com.netease.arctic.utils.SerializationUtil.simpleDeserialize(SerializationUtil.java:72) ~[amoro-core-0.6.0-SNAPSHOT.jar:?]
        ... 14 more
2023-09-14 15:29:19,316 ERROR [main] [com.netease.arctic.server.table.RuntimeHandlerChain] [] - failed to handle, ignore and continue
java.lang.NullPointerException: null
        at com.netease.arctic.server.DefaultOptimizingService$TableRuntimeHandlerImpl.doDispose(DefaultOptimizingService.java:359) ~[amoro-ams-server-0.6.0-SNAPSHOT.jar:?]
        at com.netease.arctic.server.table.RuntimeHandlerChain.doSilently(RuntimeHandlerChain.java:84) ~[amoro-ams-server-0.6.0-SNAPSHOT.jar:?]
        at com.netease.arctic.server.table.RuntimeHandlerChain.dispose(RuntimeHandlerChain.java:79) ~[amoro-ams-server-0.6.0-SNAPSHOT.jar:?]
        at com.netease.arctic.server.table.DefaultTableService.dispose(DefaultTableService.java:355) ~[amoro-ams-server-0.6.0-SNAPSHOT.jar:?]
        at com.netease.arctic.server.ArcticServiceContainer.dispose(ArcticServiceContainer.java:170) ~[amoro-ams-server-0.6.0-SNAPSHOT.jar:?]
        at com.netease.arctic.server.ArcticServiceContainer.main(ArcticServiceContainer.java:111) ~[amoro-ams-server-0.6.0-SNAPSHOT.jar:?]
```

### Anything else

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by rfyu**
## Why are the changes needed?

fix #1985 .

## Brief change log

- add a nonnull check for `com.netease.arctic.server.DefaultOptimizingService#optimizerMonitorTimer`

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (yes / no)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: ams/server/src/main/java/com/netease/arctic/server/DefaultOptimizingService.java [hunk 1]
```diff
@@ -356,7 +357,9 @@ protected void initHandler(List<TableRuntimeMeta> tableRuntimeMetaList) {
 
     @Override
     protected void doDispose() {
-      optimizerMonitorTimer.cancel();
+      if (!Objects.isNull(optimizerMonitorTimer)) {
```
Comment (shidayang) [inline ams/server/src/main/java/com/netease/arctic/server/DefaultOptimizingService.java:360]: Maybe replaced by 'Objects.noNull(optimizerMonitorTimer)'?

Comment (rfyu) [reply ams/server/src/main/java/com/netease/arctic/server/DefaultOptimizingService.java:360]: ok

**Review by baiyangtx — APPROVED** [review]
LGTM, thanks for your contribution.

**Comment by HuangFru** [comment]
Could you please complete the 'Brief change log' above? For example: "add a nonnull check for 'optimizerMonitorTimer'" or something like this.

**Comment by rfyu** [comment]
> Could you please complete the 'Brief change log' above? For example: "add a nonnull check for 'optimizerMonitorTimer'" or something like this.

ok

**Review by HuangFru — APPROVED** [review]
LGTM. Appreciate your contribution.

**Pull request merged**